core/src/mindustry/game/EventType.java

```diff
@@ -476,11 +477,12 @@ public BlockBuildEndEvent(Tile tile, @Nullable Unit unit, Team team, boolean bre
     public static class BuildRotateEvent{
         public final Building build;
         public final @Nullable Unit unit;
-        public final int previous;
+        public final int rotation, previous;
 
-        public BuildRotateEvent(Building build, @Nullable Unit unit, int previous){
+        public BuildRotateEvent(Building build, @Nullable Unit unit, int rotation, int previous){
             this.build = build;
             this.unit = unit;
+            this.rotation = rotation;
```

Comment:
What is the purpose of this? Seems redundant since this is already stored in `build.rotation`. If anything should be added to this event, it's the direction considering that it would remove the need for hacks like these (only really useful for quick rotations but still): https://github.com/mindustry-antigrief/mindustry-client/blob/b14aa9c94e756915cc0089ab9105acd93f2a9aaf/core/src/mindustry/client/utils/ClientUtils.kt#L471